src/analyses/libraryFunctions.ml

```diff
@@ -265,10 +280,12 @@ let invalidate_actions = [
     "pthread_cond_wait", readsAll; (*safe*)
     "pthread_cond_signal", readsAll;(*safe*)
     "pthread_cond_broadcast", readsAll;(*safe*)
+    "pthread_cond_destroy", readsAll;(*safe*)
```

Comment:
That function may set the passed condition variable to an invalid value, [source](https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_cond_destroy.html).

Reply:
True. I got this from our handling of `pthread_mutex_destroy`

~~~OCaml
    "pthread_mutex_destroy", readsAll;(*safe*)
 ~~~
 
I think the idea is that we use a special abstraction for mutexes anyway (distinct from the way the base analysis represents it), so even if we set this to `writesAll`, one would probably still need a separate use-after-free style analysis for mutexes, condition variables, ...

I'll make a separate issue for this.

Reply:
I actually enhanced the issue we already have now: https://github.com/goblint/analyzer/issues/200